packages/broker/test/integration/multi-account-brokers.test.ts

```diff
@@ -0,0 +1,800 @@
+        const client2Messages: Todo[] = []
+        const client3Messages: Todo[] = []
+        await Promise.all([
+            client1.resend({
+                stream: freshStreamId,
+                resend: {
+                    from: {
+                        timestamp: timeAfterFirstMessagePublished,
+                    }
+                }
+            }, (message) => {
+                client1Messages.push(message)
+            }),
+            client2.resend({
+                stream: freshStreamId,
+                resend: {
+                    from: {
+                        timestamp: timeAfterFirstMessagePublished,
+                    }
+                }
+            }, (message) => {
+                client2Messages.push(message)
+            }),
+            client3.resend({
+                stream: freshStreamId,
+                resend: {
+                    from: {
+                        timestamp: timeAfterFirstMessagePublished,
+                    }
+                }
+            }, (message) => {
+                client3Messages.push(message)
+            })
+        ])
+
+        await waitForCondition(() => client2Messages.length === 3 && client3Messages.length === 3)
+        await waitForCondition(() => client1Messages.length === 3)
+
+        expect(client1Messages).toEqual([
+            {
+                key: 2
+            },
+            {
+                key: 3
+            },
+            {
+                key: 4
+            },
+        ])
+
+        expect(client2Messages).toEqual([
+            {
+                key: 2
+            },
+            {
+                key: 3
+            },
+            {
+                key: 4
+            },
+        ])
+
+        expect(client3Messages).toEqual([
+            {
+                key: 2
+            },
+            {
+                key: 3
+            },
+            {
+                key: 4
+            },
+        ])
+    })
+
+    it('happy-path: resend range request via websocket', async () => {
+        await Promise.all([
+            client1.subscribe({
+                stream: freshStreamId
+            }, () => {}),
+            client2.subscribe({
+                stream: freshStreamId
+            }, () => {}),
+            client3.subscribe({
+                stream: freshStreamId
+            }, () => {}),
+        ])
+
+        await client1.publish(freshStreamId, {
+            key: 1
+        })
+        await wait(50)
+        const timeAfterFirstMessagePublished = Date.now()
+
+        await client1.publish(freshStreamId, {
+            key: 2
+        })
+        await wait(50)
+        await client1.publish(freshStreamId, {
+            key: 3
+        })
+        await wait(25)
+        const timeAfterThirdMessagePublished = Date.now()
+        await wait(25)
+
+        await client1.publish(freshStreamId, {
+            key: 4
+        })
+
+        await wait(1500) // wait for propagation
```

Comment:
We really need to put some kind of utility into the public client interface that waits for messages to hit storage. I have a utility that polls the `last` endpoint in the client tests but I think it should just be in the main client.

Reply:
@timoxley this whole test is a copy-paste from broker.test.ts, only using the same identities to reproduce and ensure that all those tests are passing. Not sure if there would be a more elegant solution, like providing an alternate befreEach to run the same "it" blocks as in the broker default test?